### ISSUE TITLE
Offer a button to retry checking for updates

### ISSUE THREAD

**Issue by jsoref** [rancher-sandbox/rancher-desktop#7516]
### Problem Description

I'm not precisely sure what I did, but it's possible my laptop moved between networks/was on a broken network...

Rancher Desktop 1.15.0 showed me:
![image](https://github.com/user-attachments/assets/598ca327-bbd7-48d1-bd64-ead193c8d77d)

There was no obvious way to get it to check again. (Exiting and restarting did eventually get me an update prompt, which did let me get 1.15.1 in time to be told by @jandubois that there's a 1.16.0, but that's another story.)

### Proposed Solution

Add a button to check for updates to the failed flow

### Additional Information

#1929 talks about cases where some update check fails.

**Comment by Anutrix** [comment]
Been waiting for this forever.

**Comment by VikalpRusia** [comment]
What would be the suitable place to place retry button?
- at the top beside Update available 
- at the bottom after release notes ( where apply update button appears when successful )

My suggestion would be to place it beside update available.
@jandubois , @mook-as need your inputs before I start implementing

**Comment by jsoref** [comment]
Personally, I'd be inclined to put it next to update available. But if you put it there, the button shouldn't be labeled "Retry", just "Check for updates"

But, arguably any place would be fine. Technically, it's logically more appropriate to put it after "There was an error checking for updates."

**Comment by jsoref** [comment]
Given that there's already a flow with a button below: 
<img width="1041" alt="Image" src="https://github.com/user-attachments/assets/d94d2fa7-ceb0-4055-9d28-8042c2ccc49f" />

It's probably best to put the button below.

**Comment by jandubois** [comment]
I don't understand the discussion? Why would you need a retry button if you already have the update available?

I thought you just wanted a way to force Rancher Desktop to check for updates right away instead of waiting for up to an hour when it would check again automatically. So the button would only be displayed when there are no release notes.

I think the best placement would be to move the "Check for updates automatically" checkbox to the left to make enough space behind it for a "Check now" button:

![Image](https://github.com/user-attachments/assets/b0728377-9eee-487b-a116-7bd30b5b721a)

**Comment by jandubois** [comment]
> I thought you just wanted a way to force Rancher Desktop to check for updates right away instead of waiting for up to an hour when it would check again automatically.

Rereading the top of the thread I see now that you had an error situation where the release notes have been fetched, but the update data download has failed. I don't believe this is a common scenario. Did this happen more than once (personally I have never seen it).

**Comment by jsoref** [comment]
I'm pretty sure we saw it recently on a coworker's laptop...

**Comment by VikalpRusia** [comment]
> Did this happen more than once
It happened with me once

but I agree with your suggestion @jandubois ,
Can we have check for update button which can also do retry when something fails.

**Comment by VikalpRusia** [comment]
<img width="1034" alt="Image" src="https://github.com/user-attachments/assets/64b5ff1b-ed08-411b-9a62-b97c271b847e" />
<img width="1028" alt="Image" src="https://github.com/user-attachments/assets/503a7c6c-59fa-430a-97a2-91441277dad8" />

**Comment by jsoref** [comment]
`Retry Update` isn't really correct, that would imply that _updating_ failed and that the button would attempt to update again. This is really `Recheck for available updates` or something...

**Comment by VikalpRusia** [comment]
@jsoref could you comment it on the MR itself so we can start discussion

@jandubois I had a doubt commented on the MR itself, could you clarify me please :)

**Comment by jandubois** [comment]
I don't think we need different text for failed update checks. It just makes the code more complex for no real benefit. It is also hard to test if that code actually works.

I think the button should just say "Check Now". "[X] Check for updates automatically (Check for Updates)" is less clear than "Check Now" in my opinion.

Something like this (from a different app):

![Image](https://github.com/user-attachments/assets/b1720a2a-2c9f-45fa-8458-51439b451c14)

**Comment by VikalpRusia** [comment]
@jandubois yup, seems better changed the MR.

Could you help me with this (validating) - https://github.com/rancher-sandbox/rancher-desktop/pull/8189#discussion_r1943088017

**Comment by jsoref** [comment]
Yes, `Check Now` is great.

**Comment by jandubois** [comment]
I've been able to reproduce the vanishing "Restart Now" button. I'm not exactly sure what has to be done, but it is a combination of the update already being downloaded:

![Image](https://github.com/user-attachments/assets/402525e0-cedd-4ad2-a026-e1f690f60a30)

And then disabling the "Check for updates automatically button" to prevent Rancher Desktop from updating during a restart. When you then re-enable the check after the restart (do perform the update), the "Restart Now" button may be missing:

![Image](https://github.com/user-attachments/assets/2cb07896-d3fa-4fd1-8c5f-69b779ef4054)

In re-testing this scenario the button actually re-appeared, but toggling the checkbox off and back on again made the button disappear for good.

So even if we have a "Check Now" button we may still need to fix the problem of the disappearing button.

**Comment by jsoref** [comment]
I just got this ... in my case, my computer had just done an OS update and after I logged in, I was greeted with:
<img width="1392" alt="Image" src="https://github.com/user-attachments/assets/4ee3c0b7-64eb-4e4c-bcda-961dd64110c1" />
<img width="1392" alt="Image" src="https://github.com/user-attachments/assets/d491102c-4484-4d00-a301-c445deb20fd6" />

It's possible my network wasn't up enough for rancher to do something, but had previously downloaded the release notes? 🤷‍♂ 

If there are logs you want, I'm around...
